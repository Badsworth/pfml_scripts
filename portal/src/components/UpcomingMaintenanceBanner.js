import PropTypes from "prop-types";
import React from "react";
import { Trans } from "react-i18next";
import classnames from "classnames";

/**
 * Banner displayed at the top of the site to indicate upcoming
 * scheduled maintenance.
 */
const UpcomingMaintenanceBanner = (props) => { // props.start, props.end
  const classes = classnames("upcoming-maintenance-banner bg-yellow padding-1 text-center");

  let maintenanceMessage = "";
  if(!props.start && !props.end) {
    maintenanceMessage = "components.upcomingMaintenanceBanner.messageWithNoStartAndNoEnd";
  } else if (!props.start && props.end) {
    maintenanceMessage = "components.upcomingMaintenanceBanner.messageWithEndAndNoStart";
  } else if (props.start && !props.end) {
    maintenanceMessage = "components.upcomingMaintenanceBanner.messageWithStartAndNoEnd";
  } else {
    maintenanceMessage = "components.upcomingMaintenanceBanner.messageWithStartAndEnd";
  }

  return (
    <React.Fragment>
      <div role="alert" className={classes}>
        <Trans
          i18nKey={maintenanceMessage}
          components={{
            start: props.start,
            end: props.end
          }}
        />
      </div>
    </React.Fragment>
  );
};

UpcomingMaintenanceBanner.propTypes = {
  /**
   * Start date string for the message
   **/
  start: PropTypes.string,
  /**
   * End date string for the message
   **/
  end: PropTypes.string
};

export default UpcomingMaintenanceBanner;
