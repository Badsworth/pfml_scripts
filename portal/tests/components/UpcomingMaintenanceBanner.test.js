import React from "react";
import UpcomingMaintenanceBanner from "../../src/components/UpcomingMaintenanceBanner";
import { shallow } from "enzyme";

describe("MaintenanceTakeover", () => {
  it("Renders correct content when only start date is provided", () => {
    const wrapper = shallow(<UpcomingMaintenanceBanner start="June 15, 2021, 3:00 PM EDT" end="" />);

    expect(wrapper).toMatchSnapshot();
  });

  it("Renders correct content when both start date and end date are provided", () => {
    const wrapper = shallow(<UpcomingMaintenanceBanner start="June 15, 2021, 3:00 PM EDT" end="June 17, 2021, 10:00 AM EDT" />);

    expect(wrapper).toMatchSnapshot();
  });
});
