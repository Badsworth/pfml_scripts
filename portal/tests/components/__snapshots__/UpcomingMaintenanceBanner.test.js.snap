// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaintenanceTakeover Renders correct content when both start date and end date are provided 1`] = `
<Fragment>
  <div
    className="upcoming-maintenance-banner bg-yellow padding-1 text-center"
    role="alert"
  >
    <Trans
      components={
        Object {
          "end": "June 17, 2021, 10:00 AM EDT",
          "start": "June 15, 2021, 3:00 PM EDT",
        }
      }
      i18nKey="components.upcomingMaintenanceBanner.messageWithStartEnd"
    />
  </div>
</Fragment>
`;

exports[`MaintenanceTakeover Renders correct content when only start date is provided 1`] = `
<Fragment>
  <div
    className="upcoming-maintenance-banner bg-yellow padding-1 text-center"
    role="alert"
  >
    <Trans
      components={
        Object {
          "end": "",
          "start": "June 15, 2021, 3:00 PM EDT",
        }
      }
      i18nKey="components.upcomingMaintenanceBanner.messageWithStartNoEnd"
    />
  </div>
</Fragment>
`;
